readme.md

```diff
@@ -408,14 +406,23 @@ client.get('/demo');
 
 #### got.assignOptions(parentOptions, newOptions)
 
-Extends parent options. Avoid using [object spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals) as it doesn't work recursively:
```

Comment:
I'd leave this note about avoiding `object spread` because mostly that's not what users want.

Reply:
I pruned it because I didn't think it was necessary for us to document the language feature, and the reader's attention to detail is inversely related to the length of the docs.

I can put it back if you feel strongly that it will meaningfully assist users in making the right choice.

Reply:
@sindresorhus What are your thoughts on that?

Reply:
I think it's worth keeping. I see the mistake of using Object.assign/object-spread for deep objects all the time.

Reply:
On closer inspection, I didn't actually remove this - I just inserted the function description before the warning. :-)